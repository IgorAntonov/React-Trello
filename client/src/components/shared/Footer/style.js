import styled from 'styled-components';

import { P as defaultP } from '../Typography';

export const FooterWrapper = styled.footer`
  display: flex;
  justify-content: space-between;
  align-items: center;
  
  padding: 1rem;

  background-color: #201e1c;
`;

export const P = styled(defaultP)`
  margin: 0;
  font-weight: 400;
  color: #7b7a7a;
`;

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-end;
`;

export const A = styled.a`
  margin: 0.25rem;
  padding: 0.25rem;
  
  fill: #7b7a7a;
  transition: 0.2s all;
  :hover {
    fill: #fafafa;
  }
`;
