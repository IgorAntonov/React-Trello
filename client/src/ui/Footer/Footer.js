import React from 'react';
import styled from 'styled-components';

import { Row, P } from 'Src/ui';
import { Icon } from '../Icon';

const IconBtn = styled.a`
  margin: 0.25rem;
  padding: 0.25rem;
  
  fill: #7b7a7a;
  transition: 0.2s fill;
  :hover {
    fill: #fafafa;
  }
`;

export const Footer = () => (
  <Row justify="space-between" align="center" padding="0.5rem" bgColor="#201e1c" >
    <P marginB="0" color="#7b7a7a">
      Â© 2018 Igor Antonov
    </P>
    <Row align="flex-end">
      <IconBtn href="https://github.com/IgorAntonov" >
        <Icon icon="github" width="24px" height="24px" viewBox="24" />
      </IconBtn>
      <IconBtn href="https://twitter.com/IgorAntonov18" >
        <Icon icon="twitter" width="24px" height="24px" viewBox="24" />
      </IconBtn>
      <IconBtn href="https://vk.com/id18941999" >
        <Icon icon="vk" width="24px" height="24px" viewBox="24" />
      </IconBtn>
    </Row>
  </Row>
);
