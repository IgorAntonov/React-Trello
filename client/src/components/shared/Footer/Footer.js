import React from 'react';

import { FooterWrapper, P, Wrapper, A } from './style';
import { Icon } from '../Icons';


export const Footer = () => (
  <FooterWrapper>
    <P>Â© 2018 Igor Antonov</P>
    <Wrapper>
      <div>
        <A href="https://github.com/IgorAntonov" >
          <Icon icon="github" width="24px" height="24px" viewBox="24" />
        </A>
        <A href="https://twitter.com/IgorAntonov18" >
          <Icon icon="twitter" width="24px" height="24px" viewBox="24" />
        </A>
        <A href="https://vk.com/id18941999" >
          <Icon icon="vk" width="24px" height="24px" viewBox="24" />
        </A>
      </div>
    </Wrapper>
  </FooterWrapper>
);
