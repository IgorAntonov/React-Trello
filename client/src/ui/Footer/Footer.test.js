import React from 'react';
import { shallow } from 'enzyme';

import { Footer } from './Footer';

describe('<Footer />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Footer />);
  });

  test('renders', () => {
    expect(wrapper.exists()).toBe(true);
  });
  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
