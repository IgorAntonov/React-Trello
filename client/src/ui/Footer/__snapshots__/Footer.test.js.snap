// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match snapshot 1`] = `
<styled-layout__Row
  align="center"
  bgColor="#201e1c"
  justify="space-between"
  padding="0.5rem"
>
  <P
    color="#7b7a7a"
    marginB="0"
  >
    Â© 2018 Igor Antonov
  </P>
  <styled-layout__Row
    align="flex-end"
  >
    <Footer__IconBtn
      href="https://github.com/IgorAntonov"
    >
      <Icon
        height="24px"
        icon="github"
        viewBox="24"
        width="24px"
      />
    </Footer__IconBtn>
    <Footer__IconBtn
      href="https://twitter.com/IgorAntonov18"
    >
      <Icon
        height="24px"
        icon="twitter"
        viewBox="24"
        width="24px"
      />
    </Footer__IconBtn>
    <Footer__IconBtn
      href="https://vk.com/id18941999"
    >
      <Icon
        height="24px"
        icon="vk"
        viewBox="24"
        width="24px"
      />
    </Footer__IconBtn>
  </styled-layout__Row>
</styled-layout__Row>
`;
